src/studio/src/designer/frontend/packages/schema-editor/src/components/SchemaItem.tsx

```diff
@@ -192,11 +223,14 @@ function SchemaItem(props: SchemaItemProps) {
         language={props.language}
         icon={getIconStr()}
         label={refItem ? `${item.displayName} : ${refItem.displayName}` : item.displayName}
-        onAddProperty={(item.type !== 'object') ? undefined : handleAddProperty}
+        onAddProperty={(item.type === 'object') ? handleAddProperty : undefined}
+        onAddReference={(item.type === 'object' || (item.allOf || item.anyOf || item.oneOf)) ? handleAddProperty : undefined}
+        onAddCombination={(item.type === 'object') ? handleAddProperty : undefined}
```

Comment:
Is this supposed to be the same function as on 226?

Reply:
Yeah. I found that calling the same method made sense at the time (avoids some duplication), but I'm open to suggestions if any feel otherwise.